### PR TITLE
Debugging CI

### PR DESCRIPTION
**Pull request by Smit-create**

### CODE REVIEW

**Comment by Smit-create** [comment]
ubuntu and mac are working fine.

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -30,10 +30,10 @@ jobs:
...
             python-version: "3.10"
 
-          - os: windows-latest
-            label: win-64
-            prefix: C:\Miniconda3\envs\test
-            python-version: "3.10"
+          # - os: windows-latest
+          #   label: win-64
+          #   prefix: C:\Miniconda3\envs\test
+          #   python-version: "3.10"
```
Comment (Thirumalai-Shaktivel) [inline .github/workflows/CI.yml:37]: But How?? These changes are unrelated, right!!

Comment (Smit-create) [reply .github/workflows/CI.yml]: Yeah, just debugging which platforms fails.

File: ci/environment.yml [hunk 2]
```diff
@@ -7,7 +7,7 @@ dependencies:
   - toml
   - pytest
   - jupyter
-  - xeus=1.0.1
+  - xeus
```
Comment (Smit-create) [inline ci/environment.yml:10]: Is it necessary to pin?

Comment (Thirumalai-Shaktivel) [reply ci/environment.yml]: Yes, I believe... The modules or functions used by LFortran might be taken from the xeus 1.0.1 version.

**Comment by Smit-create** [comment]
Now, Linux was working fine but some issue with mac failed the whole job. Using 3 scripts for 3 platforms should be good going as they won't affect each other's failure and might help to fix platform-specific issues. @certik  What do you think? Caching was working fine too, I checked it.

**Comment by Thirumalai-Shaktivel** [comment]
Hi @Smit-create. If you don't mind, Can I work on this PR??

**Comment by Smit-create** [comment]
Sure, you can!

**Comment by Smit-create** [comment]
Till you debug here, I'm working parallelly on separating 3 ci files.

File: .github/workflows/CI.yml [hunk 3]
```diff
@@ -42,25 +36,27 @@ jobs:
...
+        env:
+          CACHE_NUMBER: 0
         with:
           path: ${{ matrix.prefix }}
           key: ${{ matrix.label }}-conda-${{ hashFiles('ci/environment.yml') }}-${{ env.DATE }}-${{ env.CACHE_NUMBER }}
         id: cache
 
       - name: Update environment
-        run: conda env update -n test -f ci/environment.yml
-        if: steps.cache.outputs.cache-hit != 'true'
```
Comment (Thirumalai-Shaktivel) [inline .github/workflows/CI.yml:63]: I believe the issue was with this `if` statement

**Comment by Smit-create** [comment]
Thanks @Thirumalai-Shaktivel. I think we can merge this. Separating each OS in the different files can be discussed with @certik

File: .github/workflows/CI.yml (hunk 3 above)
Comment (Smit-create) [reply .github/workflows/CI.yml:63]: Yeah, I think that too. I removed it in the other PR

**Comment by certik** [comment]
I essentially reverted this in #39. We'll have to try again on this.
